File: services/template-generation.ts
import { apiFetch } from "./api-client";

export type TemplatePayload = {
  template_name: string;
  selected_template_name?: string;
  template_id?: number;
  framework?: string;
  brand_name?: string;
  industry: string;
  theme: string;
  tokens: Record<string, any>;
  structure: Record<string, any>;
};

export type TemplateGenerationRequest = {
  provider?: string;
  model?: string;
  payload: TemplatePayload;
};

export type TemplateGenerationResponse = {
  compiled_prompt: string;
};

export async function generateTemplatePrompt(
  body: TemplateGenerationRequest
): Promise<TemplateGenerationResponse> {
  return apiFetch<TemplateGenerationResponse>("/template-generation", {
    method: "POST",
    body: JSON.stringify(body),
  });
}

File: backend/app/schemas/template_generation.py
from pydantic import BaseModel
from typing import Any, Dict


class TemplatePayload(BaseModel):
    template_name: str
    selected_template_name: str | None = None
    framework: str | None = None
    industry: str
    theme: str
    brand_name: str | None = None
    template_id: int | None = None
    tokens: Dict[str, Any] = {}
    structure: Dict[str, Any] = {}
    input_config: Dict[str, Any] = {}
    component_input_configs: Dict[str, Any] = {}


class TemplateGenerationRequest(BaseModel):
    provider: str | None = None
    model: str | None = None
    payload: TemplatePayload


class TemplateGenerationResponse(BaseModel):
    compiled_prompt: str
    input_tokens: int
    output_tokens: int
    total_tokens: int


class Config:
    orm_mode = True
